src/base.jl

```diff
@@ -1491,6 +1496,26 @@ function get_buses(sys::System, bus_numbers::Set{Int})
     return buses
 end
 
+"""
+Return all the device types in the system. It does not return component types
+"""
+function get_existing_device_types(sys::System)
+    device_types = Vector{DataType}()
+    for component_type in keys(sys.data.components.data)
```

Comment:
Could you have a generator that is part of a HybridSystem that doesn't also exist as a regular component? If so, both of these functions should include `sys.data.masked_components`.

Reply:
For the purpose of this function we don't want to return the masked components. In a way this function is meant to return device types available through calling `get_components`

Reply:
Perhaps include that note in the docstring so that people in the future don't wonder whether this is an oversight?